drivers/hmis/app/graphql/mutations/ac_hmis/update_referral_posting.rb

```diff
@@ -49,12 +49,10 @@ def resolve(id:, input:)
           end
           raise ActiveRecord::Rollback if errors.any?
 
-          # This is similar to the logic in query_type.rb:can_project_accept_referral, but differs in a couple ways:
-          # - Here, we do use viewable_by when querying Enrollments. (We already threw an error above if the user doesn't have this permission)
-          # - We return information about each conflicting enrollment, so the user can fix the errors.
-          personal_ids = Hmis::Hud::Client.where(id: posting.referral.household_members.pluck(:client_id)).pluck(:PersonalID)
-          # use project_pk so that wip enrollments are included
-          conflicting_enrollments = Hmis::Hud::Enrollment.viewable_by(current_user).where(project_pk: posting.project.id, personal_id: personal_ids)
+          # Similar to query_type.rb:project_can_accept_referral, but returns info about each conflicting enrollment, so the user can fix the errors.
+          personal_ids = posting.referral.household_members.map(&:client).pluck(:PersonalID)
```

Comment:
It's safe to use the PersonalID that is on the Enrollment record. We can be sure it will match the PersonalID on the Client, because it's used in the composite key relation (see my link in my previous comment).

For example: `posting.referral.household_members.pluck(:PersonalID)`.

Reply:
It looks like the `ReferralHouseholdMember` type does not have a `PersonalID` column, only `client_id`. And looks like we can query the `Enrollments` table by this `client_id` only if we join to `clients` first. I'm not sure which one would be more performant, is it worth making a performance benchmark to find out?

Reply:
Oh I'm sorry, again! I was thinking it was enrollment.household_members, which are Enrollments. Nope this seems good!!